snowflake-schema-grant-all/main.tf.json

```diff
@@ -0,0 +1,76 @@
+{
+  "//": "\n\tNOTE: Generated by scripts/snowflake_generate_grant_all.\n\t      Changes made directly to this file will be overwritten.\n\t\t\t\tMake improvements there so everyone can benefit.",
+  "variable": {
+    "database_name": {
+      "type": "string",
+      "description": "The name of the database containing the schema on which to grant privileges.",
+      "default": null
+    },
+    "on_future": {
+      "type": "bool",
+      "description": "When this is set to true, apply this grant on all future schemas in the given database. The schema_name and shares fields must be unset in order to use on_future.",
+      "default": null
+    },
+    "per_privilege_grants": {
+      "type": "map(object({shares : list(string),roles : list(string)}))",
+      "description": "A map of privileges to authorized roles and shares. Privileges must be UPPER case.\n  This allows you to authorize extra roles/shares for specific privileges.\n  The reason this module exists is that the provider only supports one grant resource per (database_name, schema_name, table_name, on_future, with_grant_option) tuple.\n\tFor example, if you used this module to grant an ALL privilege to a role you couldn't grant a subset of the ALL privs to another role.",
+      "default": {}
+    },
+    "roles": {
+      "type": "set(string)",
+      "description": "Grants privilege to these roles.",
+      "default": null
+    },
+    "schema_name": {
+      "type": "string",
+      "description": "The name of the schema on which to grant privileges.",
+      "default": null
+    },
+    "shares": {
+      "type": "set(string)",
+      "description": "Grants privilege to these shares (only valid if on_future is unset).",
+      "default": null
+    },
+    "with_grant_option": {
+      "type": "bool",
+      "description": "When this is set to true, allows the recipient role to grant the privileges to other roles.",
+      "default": null
+    }
+  },
+  "locals": {
+    "privileges": [
+      "CREATE EXTERNAL TABLE",
+      "CREATE FILE FORMAT",
+      "CREATE FUNCTION",
+      "CREATE MASKING POLICY",
+      "CREATE MATERIALIZED VIEW",
+      "CREATE PIPE",
+      "CREATE PROCEDURE",
+      "CREATE SEQUENCE",
+      "CREATE STAGE",
+      "CREATE STREAM",
+      "CREATE TABLE",
+      "CREATE TASK",
+      "CREATE TEMPORARY TABLE",
+      "CREATE VIEW",
+      "MODIFY",
+      "MONITOR",
+      "OWNERSHIP",
+      "USAGE"
+    ]
+  },
+  "resource": {
+    "snowflake_schema_grant": {
+      "all": {
```

Comment:
⚠️ **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing version constraint for provider "snowflake" in "required_providers"